SignalMessaging/utils/ThreadUtil.m

```diff
@@ -247,7 +249,8 @@ + (ThreadDynamicInteractions *)ensureDynamicInteractionsForThread:(TSThread *)th
 
     ThreadDynamicInteractions *result = [ThreadDynamicInteractions new];
 
-    [dbConnection readWriteWithBlock:^(YapDatabaseReadWriteTransaction *transaction) {
+    NSMutableArray *writeBlocks = [NSMutableArray new];
+    [uiDatabaseConnection readWithBlock:^(YapDatabaseReadTransaction *transaction) {
```

Comment:
We now read from the uiDatabaseConnection, and queue any reads to happen in a single transaction on the editing db connection.